src/adr/0034-distributed-listeners-configuration.md

```diff
@@ -0,0 +1,106 @@
+bindings.
+
+This event sourcing system differs from the one defined in
+[26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md) by the
+fact that we should also keep track of listener configuration.
+
+Upon start, James will ensure the **configured mailbox listener event sourcing system** contains the guice injected 
+listeners, and add them if missing (handling the RabbitMQ bindings by this mean), then starts the eventBus which will
+consume the given queues.
+
+If a listener is configured with a class unknown to James, the start-up fails. This can happen if a custom jar, containing
+user implemented mailbox listeners is partially deployed.
+
+This differs from [26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md)
+by the fact we no longer need to register all listeners at once.
+
+A WebAdmin endpoint will allow:
+
+ - **to add a listener** to the one configured. Such a call:
+    - Will fail if the listener is unknown to James, or if the corresponding group is unknown.
+    - Upon success the listener is added to the **configured mailbox listener aggregate**, and the listener is 
+    registered locally. No broadcast is attempted, meaning that other James servers will need a reboot to actually start 
+    consuming the queue.
+ - **to remove a listener**. Such a call:
+    - Will fail if the listener is required by Guice bindings on the current server (distributed check will not be 
+    implemented) or if the listener is not configured.
+    - Upon success, the listener is removed from the **configured mailbox listener aggregate**, and the listener is 
+    unregistered locally. No broadcast is attempted, meaning that other James servers will need a reboot to actually stop 
+    consuming the queue. However, new events will stop arriving in the queue as its binding will be removed.
+
+Integration tests relying on additional mailbox listeners of the distributed James product will require to be ported to 
+perform additional mailbox listener registry with this WebAdmin endpoint. JMAP SpamAssassin, quota mailing tests are 
+concerned.
+
+We will also expose an endpoint listing the groups currently in use, and for each group the associated configuration, if 
+any. This will query the **configured mailbox listener aggregate**.
+
+We will introduce a health check to actually ensure that RabbitMQ resources match the configured listeners, and propose
+a WebAdmin endpoint to add/remove bindings/queue in a similar fashion of what had been proposed in 
+[26. Removing a configured additional MailboxListener](0026-removing-configured-additional-mailboxListeners.md).
+
+## Consequences
+
+All products other than "Distributed James" are unchanged.
+
+All the currently configured additional listeners will need to be registered.
+
+The definition of mailbox listeners is thus centralized and we are not exposed to an heterogeneous configuration 
```

Comment:
It would make sense to describe how someone deploy a new listener with this implementation

Reply:
Sample rest API calls is IMO too low level for and ADR.

Reply:
I'm not saying we should talk about REST but rather a "user story" with steps

Reply:
Given what I had written calling the webadmin endpoint should be enough?

Reply:
> Given what I had written calling the webadmin endpoint should be enough?

I would not ask for more.

The steps are:
1. create a jar with your listener
2. deploy that jar to every james server of the cluster
3. restart (or not?) servers
4. call webadmin to enable the listener (with the conf or not?)

Same about disabling a listener

Reply:
Ok, your question was missing `custom` as a prefix of `new listener`.